src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -1508,5 +1508,168 @@ public void nameof(string x)
             var option = TestOptions.ReleaseDll;
             CreateCompilation(source, options: option).VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(40229, "https://github.com/dotnet/roslyn/issues/40229")]
+        public void TestCanReferenceInstanceMembersFromStaticMemberInNameof()
+        {
+            var source = @"
+public class C
+{
+    public string S { get; }
+    public static string M() => nameof(S.Length);
```

Comment:
Consider verifying code execution for the positive cases, so we can see what it prints.